src/server/worker/datum/datum.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package datum;
+option go_package = "github.com/pachyderm/pachyderm/src/server/worker/datum";
+
+import "gogoproto/gogo.proto";
+
+import "client/pps/pps.proto";
+import "server/worker/common/common.proto";
+
+enum State {
+  PROCESSED = 0;
+  FAILED = 1;
```

Comment:
Can we make these `SUCCESS` and `FAILURE` to match with the states for jobs and pipelines?

Reply:
Sure.

Reply:
@jdoliner Actually, these names don't seem to fit well with the other names for pipeline, job, and datum states. It looks we are mixing nouns and verbs rather than following one pattern (i.e. for jobs starting, running, egressing, and killed are verbs while success and failure are nouns). I think it might be better to stick with the verb forms across the board because it reads better, at least in my opinion. So we could have succeeded and failed across the board rather than success and failure.

Reply:
Hmm, I agree with your logic that we shouldn't be mixing verbs and nouns. If we're going to change it though we should change it for all of them. So I think we should probably make this PR agree with the existing names and then switch all of them over before we release 2.0.